src/raftstore/store/peer.rs

```diff
@@ -2055,4 +2085,139 @@ mod tests {
             }
         }
     }
+
+    #[allow(useless_vec)]
+    #[test]
+    fn test_request_inspector() {
+        struct DummyInspector {
+            applied_index_term: bool,
+            lease_state: LeaseState,
+        }
+        impl RequestInspector for DummyInspector {
+            fn has_applied_current_term(&self) -> bool {
+                self.applied_index_term
+            }
+            fn inspect_lease(&mut self) -> LeaseState {
+                self.lease_state.clone()
+            }
+        }
+
+        let mut table = vec![];
+
+        // Ok(_)
+        let mut req = RaftCmdRequest::new();
+        let mut admin_req = raft_cmdpb::AdminRequest::new();
+
+        req.set_admin_request(admin_req.clone());
+        table.push((req.clone(), RequestPolicy::ProposeNormal));
+
+        admin_req.set_change_peer(raft_cmdpb::ChangePeerRequest::new());
+        req.set_admin_request(admin_req.clone());
+        table.push((req.clone(), RequestPolicy::ProposeConfChange));
+        admin_req.clear_change_peer();
+
+        admin_req.set_transfer_leader(raft_cmdpb::TransferLeaderRequest::new());
+        req.set_admin_request(admin_req.clone());
+        table.push((req.clone(), RequestPolicy::ProposeTransferLeader));
+        admin_req.clear_transfer_leader();
+        req.clear_admin_request();
+
+        for (op, policy) in vec![
+            (CmdType::Get, RequestPolicy::ReadLocal),
+            (CmdType::Snap, RequestPolicy::ReadLocal),
+            (CmdType::Put, RequestPolicy::ProposeNormal),
+            (CmdType::Delete, RequestPolicy::ProposeNormal),
+            (CmdType::DeleteRange, RequestPolicy::ProposeNormal),
+            (CmdType::IngestSST, RequestPolicy::ProposeNormal),
+        ] {
+            let mut request = raft_cmdpb::Request::new();
+            request.set_cmd_type(op);
+            req.set_requests(vec![request].into());
+            table.push((req.clone(), policy));
+        }
+
+        for b in vec![true, false] {
```

Comment:
How about list them explicitly? Seems only list the read requests is enough.

Reply:
I think it's fine. It also improves coverage.

Reply:
How can they improve coverage as they always hit the same line? List them explicitly can make tests more clear, so reader don't need to think if the transformation rule is correct or not.

Reply:
After listing them one by one, the tests becoms to long, instead, I will add some comments to make it more clear.